observer/observer.go

```diff
@@ -202,6 +207,34 @@ func (o *Observer) IsAnyJobRunning(repository string) bool {
 	return false
 }
 
+// IsLimitConcurrentJobsReached checks if the limit of concurrent jobs by type (backup, check, etc)
+// has been reached
+func (o *Observer) IsLimitConcurrentJobsReached(jobType cfg.JobType, limit int, repository string) bool {
+	listOfJobs := o.GetJobsByRepository(repository)
+	if limit > 0 {
+		if len(listOfJobs) == 0 {
+			return false
+		}
+
+		result := filter(listOfJobs, func(val ObservableJob) bool {
+			return val.JobType == jobType
+		})
+
+		return len(result) > limit
+	}
+	return false
+}
+
+func filter(arr []ObservableJob, cond func(ObservableJob) bool) []ObservableJob {
+	result := []ObservableJob{}
+	for i := range arr {
+		if cond(arr[i]) {
+			result = append(result, arr[i])
+		}
+	}
+	return result
+}
```

Comment:
I think this can be simplified with an early exit:
```suggestion
func (o *Observer) IsLimitConcurrentJobsReached(jobType cfg.JobType, limit int, repository string) bool {
	if limit <= 0 {
		return false
	}
	listOfJobs := o.GetJobsByRepository(repository)
	if len(listOfJobs) == 0 {
		return false
	}

	count := 0
	for _, job := range listOfJobs {
		if job.JobType == jobType {
			count++
			if count >= limit {
				return true
			}
		}
	}
	return false
}
```